schema/messaging.proto

```diff
@@ -0,0 +1,508 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "google/api/annotations.proto";
+import "google/longrunning/operations.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+import "google/rpc/error_details.proto";
+
+package google.showcase.v1alpha3;
+
+option go_package = "github.com/googleapis/gapic-showcase/server/genproto";
+option java_package = "com.google.showcase.v1alpha3";
+option java_multiple_files = true;
+
+option (google.api.metadata) = {
+  product_uri: "https://github.com/googleapis/gapic-showcase"
+};
+
+// This resource is defined in google/showcase/v1alpha3/user.proto. Since
+// google/showcase/v1alpha3/user.proto is not imported from this file, the
+// resource needs to be defined here to be used in this API.
+option (google.api.resource_definition) = {
```

Comment:
Oh, nice. This is a surprise benefit of @chrisdunelm's work that I had not considered -- stubbed resources (which are totally a thing).

Reply:
Agreed!

Reply:
Nope. It is needed because `user.proto` is not imported from `messaging.proto` thus the resource definition is not guaranteed to be loaded from user.proto.

Reply:
Oh. Hmm. Interesting.

Why is user.proto not imported from messaging.proto? Is `Identity` not needed there?

Reply:
Correct, Identity is not needed. But for actual usage of the api, the flow would be something like the following which is why I created the identity proto. 

```
user = Identity.CreateUser()
room = Messaging.CreateRoom()
messaging.CreateMessage(room, {user: user, text: text})

Reply:
Ah, I see.

Something to think about: I think it is reasonable to expect an entire package to be generated together (and for errors to happen if someone tries to generate only parts of a package). We should probably add a note about that in our documentation.

This is still fine as-is, however. _Stet._